src/modules/Message/utils/tokens/tokenize.ts

```diff
@@ -86,6 +100,64 @@ export function identifyUrlsAndStrings(token: Token[]): Token[] {
   return results;
 }
 
+/**
+ * For every string token in the given list of tokens, if the token contains markdowns, the token is split into
+ * string tokens and markdown tokens in the original order.
+ * Returns a new array tokens.
+ * @param tokens
+ */
+export function splitTokensWithMarkdowns(tokens: Token[]): Token[] {
+  const prevTokens = tokens;
+  const newTokens = [];
+  prevTokens.forEach((token) => {
+    if (token.type === TOKEN_TYPES.mention || token.type === TOKEN_TYPES.markdown) {
+      newTokens.push(token);
+      return;
+    }
+    const rawStr = token.value;
+    // @ts-ignore
+    const matches = [...rawStr.matchAll(MarkdownRegex)];
```

Comment:
Do we have a reason for copying `rawStr.matchAll(MarkdownRegex)` and create a new array here?

Reply:
purpose was not to copy but to create an array from iterator